[android, docs] Clean up MapSnapshotter javadoc.

ChrisLoer opened the pull request:
- Make internal/native methods private
- Add descriptions, example.
- Push users towards expected configuration options of Size/Style/Camera
- Specify in docs "don't modify options while snapshot is in progress" (see issue #11825)
- Specify in docs that starting a snapshot while one is in progress is an error
- Emphasize that "region" for the most part clobbers "camera position" (it's not just the target that can change)
- Update `MapboxMap#snapshot` with a suggestion to use `MapSnapshotter` instead.

The motivation here is that I've been doing some `MapSnapshotter` concurrency debugging, and just from looking at the docs I had a hard time figuring out how the Android wrapper was meant to be used. This PR is mainly just a bunch of small tweaks related to things that confused me.

Their shouldn't be any change in behavior in this PR (although several methods do transition from protected->private, technically a change...). I think it's weird that we include `finalize` in the docs (because we have to make it protected to expose it for the native wrapper), but it looks like we follow that pattern for other wrapper classes.

Apologies that I'm a javadoc noob and I don't really know what conventions to follow.

/cc @tobrun @zugaldia @ivovandongen

ChrisLoer commented:
> Emphasize that "region" for the most part clobbers "camera position" (it's not just the target that can change)

Experimenting a little further, I see that bearing and angle remain unchanged, although zoom can change as well as center position. I'm not sure if it's better to get into that detail or just leave it at "overridden" (which would hopefully push people towards using a single explicit camera position, which will have more predictable behavior).

tobrun commented on platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java:
```diff
@@ -2047,7 +2047,10 @@ public OnInfoWindowCloseListener getOnInfoWindowCloseListener() {
 
   /**
-   * Takes a snapshot of the map.
+   * Renders a snapshot of the current map state as a {@link Bitmap}.
```
- we tend to not use any `{@link .. }` in the first line of the javadoc, this is to avoid generating links directly in overview pages. 
 - we tend to not link against Android SDK classes (scope for linking is only within the SDK)

tobrun commented on platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java:
```diff
@@ -29,22 +29,24 @@
- * off the UI thread. The snapshotter itself must be used on
- * the UI thread (for access to the main looper)
+ * Use MapSnapshotter to create static map images, rendered in the background.
+ * <br>
```
the usage of <br> has led to javadoc generation issues in the past, we tend to use `<p> ... </p>` instead

tobrun commented on platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java:
```diff
@@ -79,7 +79,11 @@
   /**
-   * MapSnapshotter options
+   * Specifies the parameters for drawing a static map image using MapSnapshotter.
+   * <br><br>
```
same <p> remark as above

tobrun commented on platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java:
```diff
@@ -283,7 +296,7 @@ public void cancel() {
    * @param mapSnapshot the map snapshot to draw the overlay on
    */
-  protected void addOverlay(MapSnapshot mapSnapshot) {
+  private void addOverlay(MapSnapshot mapSnapshot) {
```
End developers use this method to draw image overlays between the map and our logo / attribution. Eg. a user wants to show a weather map on top of a map image. To make this work with logo / attribution, they will have to override this method, draw the overlay and call into super.addOverlay to add the logo/attribution. Would prefer keeping this protected to support those cases.

ChrisLoer replied:
Oops, I thought I knew that! Not sure why I made this mistake...

ChrisLoer commented:
Thanks for the review! Addressed the comments and also tightened the "camera position override" language back to "camera position target override".... which probably doesn't mean much to the average person reading the docs, but is a bit more precise about what's happening. I'm hoping they'll just want to set the camera position directly. 🤷‍♂️
